packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -58,8 +58,10 @@ export async function handleHotUpdate(
 	}
 }
 
-const frontmatterRE = /^\-\-\-.*?^\-\-\-/ms;
+const frontmatterRE = /^---.+?^---/ms;
+// eslint-disable-next-line regexp/no-super-linear-backtracking
 const scriptRE = /<script(?:\s.*?)?>.*?<\/script>/gs;
+// eslint-disable-next-line regexp/no-super-linear-backtracking
```

Comment:
We should explain why it's safe to suppress the rule

Reply:
I added these mainly because I don't know how to fix it 😅 From what I understand, it seems to be flagging a potential perf and ReDos issue, but it's not too bad as it's only processing on source code and not user input. So I'm not sure adding a comment that we don't know how to fix would be helpful.

Reply:
I believe every comment is useful, even the ones that say: "we don't know how to fix it" :)

Reply:
I added them 👍